crowd_anki/representation/deck.py

```diff
@@ -126,30 +126,27 @@ def _load_metadata_from_json(self, json_dict):
 
         self.metadata = DeckMetadata(new_deck_configs, new_models)
 
-    def save_to_collection(self,
-                           collection,
-                           import_config: ImportConfig,
-                           parent_name="",
-                           save_configs=True,
-                           save_note_models=True,
-                           model_map_cache=None):
-        if save_configs:  # Todo when update implemented multiple save can be harmless and code simpler
-            for config in self.metadata.deck_configs.values():
-                config.save_to_collection(collection)
-
-        if save_note_models:
-            for note_model in self.metadata.models.values():
-                note_model.save_to_collection(collection, import_config.use_note_models)
+    def save_to_collection(self, collection, import_config: ImportConfig):
+        for config in self.metadata.deck_configs.values():
+            config.save_to_collection(collection)
 
+        for note_model in self.metadata.models.values():
+            note_model.save_to_collection(collection)
+
+        self.save_children_and_notes(collection=collection,
+                                     import_config=import_config,
+                                     parent_name="",
+                                     model_map_cache=defaultdict(dict))
+
+    def save_children_and_notes(self, collection, import_config: ImportConfig, parent_name, model_map_cache):
         name = self._save_deck(collection, parent_name)
```

Comment:
this should still be part of main function though

Reply:
ah, it does not quite work in that case. the method name is misleading in this case though, as it saves the deck itself as-well

Reply:
Yes it does also save the deck, hmm. Alright, just changed the name to `save_deck_children_and_notes` to be more clear. If you have a better suggestion lemme know :+1:

Reply:
Oh nevermind, you already fixed it :laughing: